internal/pkg/addons/storage.go

```diff
@@ -0,0 +1,118 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+// Package addons contains the service to manage addons.
+package addons
```

Comment:
Sorry, this is my mistake but would you mind renaming the package to `addon` instead of `addons`?
https://rakyll.org/style-packages/#no-plurals

It will make it nice to read `addon.DynamoDB` vs `addons.DynamoDB`.